app/src/userpages/components/DataUnionPage/Item.jsx

```diff
@@ -282,13 +288,20 @@ type Props = {
 const Item = ({ product, stats }: Props) => {
     const { copy } = useCopy()
     const isMounted = useIsMounted()
+    const productId = product && product.id
+    const productName = product && product.name
+    const dataUnionId = product && product.beneficiaryAddress
+
     const [isOpen, setIsOpen] = useState(false)
     const [subscriberCount, setSubscriberCount] = useState(false)
     const [dataUnion, setDataUnion] = useState(null)
-    const { wrap: wrapSubscriberLoad, isPending: loadingSubscriberCount } = usePending('dataunion.item.SUBSCRIBERS')
-    const { wrap: wrapDataUnionLoad, isPending: loadingDataUnion } = usePending('dataunion.item.DATAUNION')
-    const { wrap: wrapJoinRequestLoad, isPending: loadingJoinRequests } = usePending('dataunion.item.JOINREQUESTS')
-    const loading = loadingSubscriberCount || loadingDataUnion || loadingJoinRequests
+    const { wrap: wrapSubscriberLoad, isPending: loadingSubscriberCount } = usePending(`dataunion.item.${productId || ''}.SUBSCRIBERS`)
+    const { wrap: wrapDataUnionLoad, isPending: loadingDataUnion } = usePending(`dataunion.item.${productId || ''}.DATAUNION`)
```

Comment:
Pending context is shared within the whole list, added here the product specifier